redisvl/schema/validation.py

```diff
@@ -0,0 +1,275 @@
+                    field_name
+                )
+
+            # Register validators for VECTOR fields
+            elif field.type == FieldTypes.VECTOR:
+                dims = field.attrs.dims  # type: ignore
+                datatype = field.attrs.datatype  # type: ignore
+
+                def make_vector_validator(
+                    fname: str, dims: int, datatype: VectorDataType
+                ):
+                    @field_validator(fname, mode="after")
+                    def _validate_vector(cls, value):
+                        # Skip validation for None values
+                        if value is not None:
+                            # Handle list representation
+                            if isinstance(value, list):
+                                # Validate dimensions
+                                if len(value) != dims:
+                                    raise ValueError(
+                                        f"Vector field '{fname}' must have {dims} dimensions, got {len(value)}"
+                                    )
+                                # Validate data types
+                                datatype_str = str(datatype).upper()
+                                # Integer-based datatypes
+                                if datatype_str in ("INT8", "UINT8"):
+                                    # Check range for INT8
+                                    if datatype_str == "INT8":
+                                        if any(v < -128 or v > 127 for v in value):
+                                            raise ValueError(
+                                                f"Vector field '{fname}' contains values outside the INT8 range (-128 to 127)"
+                                            )
+                                    # Check range for UINT8
+                                    elif datatype_str == "UINT8":
+                                        if any(v < 0 or v > 255 for v in value):
+                                            raise ValueError(
+                                                f"Vector field '{fname}' contains values outside the UINT8 range (0 to 255)"
+                                            )
+                        return value
+
+                    return _validate_vector
+
+                class_dict[f"validate_{field_name}"] = make_vector_validator(
+                    field_name, dims, datatype
+                )
+
+        # Create class dictionary with annotations and field metadata
+        class_dict.update(
+            **{
+                "__annotations__": annotations,
+                "model_config": {"arbitrary_types_allowed": True, "extra": "allow"},
+            }
+        )
+
+        # Create the model class using type()
+        model_name = f"{schema.index.name}__PydanticModel"
+        return type(model_name, (BaseModel,), class_dict)
+
+
+def extract_from_json_path(obj: Dict[str, Any], path: str) -> Any:
```

Comment:
Hmmmm. I'd use a formal JSONPath parser here.

Reply:
Yeah fair, I've not used the lib much. If you have cycles today to timebox this, I would welcome that contribution!